adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -238,7 +240,7 @@ protected void uploadMessage(final AmqpContext context) {
             return Future.succeededFuture();
         }).otherwise(t -> {
             final ClientErrorException error = (ClientErrorException) t;
-            context.reject(error);
+            context.reject(AmqpError.NOT_ALLOWED, error);
```

Comment:
It seems like the ClientErrorException will always contain a 400 status code. Do we really want to map this code to AMQP's not-allowed error? The AMQP spec defines `amqp:not-allowed` as
> The peer tried to use a frame in a manner that is inconsistent with the semantics defined in
the specification.

This doesn't seem to be a good match, does it? Maybe we should create our own error here. What about `hono:bad-request`?

Reply:
yes, i thought so too. thanks